xray/src/inpaint.rs

```diff
@@ -0,0 +1,262 @@
+// Inpainting is applied to fill holes in the X-ray quadtree leaf tiles.
+// As inpainting depends on sampling the neighborhood of pixels, if we
+// directly took the original tiles as input, we would get visible color
+// differences for the inpainted pixels along the tile borders. Because
+// of this we enlarge all tiles, so they overlap each other by half of their
+// dimension, apply inpainting on the enlarged tiles and bilinearly interpolate
+// between tiles, before cutting out the original tile.
+
+use crate::utils::{get_image_path, image_from_path, interpolate_subimages};
+use fnv::FnvHashSet;
+use image::{DynamicImage, GenericImage, GenericImageView, ImageResult, Luma, Rgba, RgbaImage};
+use imageproc::distance_transform::Norm;
+use imageproc::map::{map_colors, map_colors2};
+use imageproc::morphology::close;
+use point_viewer::color::TRANSPARENT;
+use point_viewer::utils::create_syncable_progress_bar;
+use quadtree::{Direction, NodeId, SpatialNodeId};
+use scoped_pool::Pool;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use texture_synthesis::{Dims, Example, Session};
+
+fn inpaint(image: RgbaImage, distance_px: u8) -> RgbaImage {
```

Comment:
I think it would help if we added few comments with short explanations what some of these lines do.

Reply:
Done.